server/emails/templates/DocumentMentionedEmail.tsx

```diff
@@ -26,13 +37,64 @@ export default class DocumentMentionedEmail extends BaseEmail<
   InputProps,
   BeforeSend
 > {
-  protected async beforeSend({ documentId }: InputProps) {
+  protected async beforeSend({ documentId, revisionId, userId }: InputProps) {
     const document = await Document.unscoped().findByPk(documentId);
     if (!document) {
       return false;
     }
 
-    return { document };
+    let currDoc: Document | Revision = document;
+    let prevDoc: Revision | undefined;
+
+    if (revisionId) {
+      const revision = await Revision.findByPk(revisionId);
+      if (!revision) {
+        return false;
+      }
+      currDoc = revision;
+      prevDoc = (await revision.before()) ?? undefined;
+    }
+
+    const currMentions = DocumentHelper.parseMentions(currDoc, {
+      type: "user",
+      modelId: userId,
+    });
+    const prevMentions = prevDoc
+      ? DocumentHelper.parseMentions(prevDoc, {
+          type: "user",
+          modelId: userId,
+        })
+      : [];
+
+    const firstNewMention = differenceBy(currMentions, prevMentions, "id")[0];
+    if (!firstNewMention) {
+      return false;
```

Comment:
I don't think this early return or the one below are ideal, as they will potentially result in an @mention not generating an email. It would be better to just fallback to the existing case of not having the preview if we cannot for some reason gather the correct document context